plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,291 @@
+		resp.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+		err = gatherWeatherUrl(bufio.NewReader(resp.Body), tags, acc)
+		return err
+	default:
+		return fmt.Errorf("%s returned unexpected content type %s", addr.String(), contentType)
+	}
+}
+
+type WeatherEntry struct {
+	Dt     int64  `json:"dt"`
+	Dttxt  string `json:"dt_txt"` // empty for weather/
+	Clouds struct {
+		All int64 `json:"all"`
+	} `json:"clouds"`
+	Main struct {
+		GrndLevel float64 `json:"grnd_level"` // empty for weather/
+		Humidity  int64   `json:"humidity"`
+		SeaLevel  float64 `json:"sea_level"` // empty for weather/
+		Pressure  float64 `json:"pressure"`
+		Temp      float64 `json:"temp"`
+		TempMax   float64 `json:"temp_max"`
+		TempMin   float64 `json:"temp_min"`
+	} `json:"main"`
+	Rain struct {
+		Rain3 float64 `json:"3h"`
+	} `json:"rain"`
+	Sys struct {
+		Pod     string  `json:"pod"`
+		Country string  `json:"country"`
+		Message float64 `json:"message"`
+		Id      int64   `json:"id"`
+		Type    int64   `json:"type"`
+		Sunrise int64   `json:"sunrise"`
+		Sunset  int64   `json:"sunset"`
+	} `json:"sys"`
+	Wind struct {
+		Deg   float64 `json:"deg"`
+		Speed float64 `json:"speed"`
+	} `json:"wind"`
+	Weather []struct {
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+		Id          int64  `json:"id"`
+		Main        string `json:"main"`
+	} `json:"weather"`
+
+	// Additional entries for weather/
+	Id    int64  `json:"id"`
+	Name  string `json:"name"`
+	Coord struct {
+		Lat float64 `json:"lat"`
+		Lon float64 `json:"lon"`
+	} `json:"coord"`
+	Visibility int64 `json:"visibility"`
+}
+
+type Status struct {
+	City struct {
+		Coord struct {
+			Lat float64 `json:"lat"`
+			Lon float64 `json:"lon"`
+		} `json:"coord"`
+		Country string `json:"country"`
+		Id      int64  `json:"id"`
+		Name    string `json:"name"`
+	} `json:"city"`
+	List []WeatherEntry `json:"list"`
+}
+
+func gatherStatusUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	if err := dec.Decode(status); err != nil {
+		return fmt.Errorf("Error while decoding JSON response: %s", err)
+	}
+	status.Gather(tags, acc)
+	return nil
+}
+
+func gatherWeatherUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	e := WeatherEntry{}
+	if err := dec.Decode(&e); err != nil {
+		return fmt.Errorf("Error while decoding JSON response: %s", err)
+	}
+	status.List = make([]WeatherEntry, 0)
+	if len(e.Name) > 0 {
+		status.List = append(status.List, e)
+		status.City.Coord.Lat = e.Coord.Lat
+		status.City.Coord.Lon = e.Coord.Lon
+		status.City.Id = e.Id
+		status.City.Name = e.Name
+		status.Gather(tags, acc)
+	}
+	return nil
+}
+
+func (s *Status) Gather(tags map[string]string, acc telegraf.Accumulator) {
+	tags["city_id"] = strconv.FormatInt(s.City.Id, 10)
+	for _, e := range s.List {
+		tm := time.Unix(e.Dt, 0)
```

Comment:
> This appears to be the timestamp being forecasted, as opposed to the time the forecast was made. As such, we probably shouldn't use it as the timestamp, since this will make it impossible to track changes in the forecast over time since the forecast will be overwritten each time.

It's intentional: if you visualize the data it's useful to always have the most up to date info on screen.

One way we can compare the changes in the forecast without losing information is tagging. If we tag each data point returned by the API:

- First point (+3h) is tagged `forecast=3h`
- Second point (+6h) is tagged `forecast=6h`
- etc

Eventually, one can select the current weather, the `forecast=24h`, and `forecast=48h`, this makes three time series that are easy to visualize and compare.

Reply:
Changes in the last commit. I'm running the plugin for a couple of hours to show the results.

Reply:
This seems like a really good solution, makes it very easy to line up all the data for a forecasted time.